symfony/plugins/orangehrmAdminPlugin/Dao/WorkShiftDao.php

```diff
@@ -101,24 +136,125 @@ public function getWorkShiftEmployeeIdList() {
             }
 
             return $employeeIds;
-
             // @codeCoverageIgnoreStart
         } catch (Exception $e) {
             throw new DaoException($e->getMessage(), $e->getCode(), $e);
         }
         // @codeCoverageIgnoreEnd
     }
 
-    public function saveEmployeeWorkShiftCollection(Doctrine_Collection $empWorkShiftCollection) {
+    public function saveEmployeeWorkShiftCollection(Doctrine_Collection $empWorkShiftCollection)
+    {
         // TODO
         try {
-
             $empWorkShiftCollection->save();
-
             // @codeCoverageIgnoreStart
         } catch (Exception $e) {
             throw new DaoException($e->getMessage(), $e->getCode(), $e);
         }
         // @codeCoverageIgnoreEnd
     }
+
+    /**
+     * @param WorkShift $workShift
+     * @param array $empNumber
+     * @return WorkShift
+     */
+    public function saveWorkShift(WorkShift $workShift, array $empNumber): WorkShift
+    {
+        $this->persist($workShift);
+        if (sizeof($empNumber) > 0) {
+            // this function will invoke only if the array have some values
+            $this->saveEmployeeWorkShift($empNumber, $workShift);
+            return $workShift;
+        }
+        return $workShift;
+    }
+
+    /**
+     * @param array $empNumber
+     * @param WorkShift $workShift
+     */
+    public function saveEmployeeWorkShift(array $empNumber, WorkShift $workShift): void
+    {
+        foreach ($empNumber as $empNo) {
+            $employeeWorkShift = new EmployeeWorkShift();
+            $employee = $this->getRepository(Employee::class)->find($empNo);
+            $employeeWorkShift->setWorkShift($workShift);
+            $employeeWorkShift->setEmployee($employee);
+            $this->persist($employeeWorkShift);
```

Comment:
this is bit costly,
refer:
https://github.com/orangehrm/orangehrm/blob/5.0/symfony/plugins/orangehrmLeavePlugin/Dao/LeaveEntitlementDao.php#L224-L234

https://www.doctrine-project.org/projects/doctrine-orm/en/2.9/reference/batch-processing.html#batch-processing

Reply:
done